providers/infrastructure-vsphere/v1.5.1/cconly/base.yaml

```diff
@@ -2100,3 +2103,56 @@ spec:
             {{- range .additionalFQDN }}
             - {{ . }}
             {{- end }}
+  - name: systemWideProxy
+    enabledIf: '{{ .proxy.systemWide }}'
+    definitions:
+    - selector:
+        apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+        kind: KubeadmControlPlaneTemplate
+        matchResources:
+          controlPlane: true
+      jsonPatches:
+      - op: add
+        path: /spec/template/spec/kubeadmConfigSpec/preKubeadmCommands/-
+        value: echo "export HTTP_PROXY={{- .proxy.httpProxy }}" >> /etc/profile
+      - op: add
+        path: /spec/template/spec/kubeadmConfigSpec/preKubeadmCommands/-
+        value: echo "export HTTPS_PROXY= {{- .proxy.httpsProxy }}" >> /etc/profile
+      - op: add
+        path: /spec/template/spec/kubeadmConfigSpec/preKubeadmCommands/-
+        value: echo "export NO_PROXY={{- .proxy.noProxy }}" >> /etc/profile
```

Comment:
No proxy value should be processed like this
```
echo 'export NO_PROXY={{- list "localhost" "127.0.0.1" ".svc" ".svc.cluster.local" ((list "IPv6" "DualStack" | has .builtin.cluster.network.ipFamily) | ternary  "::1" nil) | concat .proxy.noProxy .builtin.cluster.network.services .builtin.cluster.network.pods | uniq | sortAlpha | join "," }}' >> /etc/profile
```

https://github.com/vmware-tanzu/tanzu-framework/compare/topic/lubron/system-wide-proxy?expand=1